doc/devel/gitwash/matplotlib_for_dev.rst

```diff
@@ -0,0 +1,20 @@
+.. _matplotlib-for-dev:
+
+=================================================
+Installing matplotlib from source for development
+=================================================
+
+After obtaining a local copy of the matpotlib source code (:ref:`set-up-fork`),
+navigate to the matplotlib directory and run the following in the shell:
+
+::
+    
+    python setup.py develop
```

Comment:
@QuLogic: Try e.g. checking out the matplotlib 1.5 branch, doing an editable install, and then after that switch to the master branch for more hacking: `git checkout master`. Now pip thinks your virtualenv has 1.5 (check out the metadata files in the .egg-info directory -- these only get regenerated if you re-do the editable install), but it actually has a mix of master's .py files and 1.5's extension modules. Rebuild the extension modules alone, and now you have a consistent installation of master (2.x) but pip still thinks it's 1.5. Then pip install some package that depends on `matplotlib (>=2.0)` and pip will freak out and try to uninstall your dev version, because it isn't good enough, even though it is, etc. You can make it work, but it's intrinsically flaky and you have to be constantly aware of all the weird pitfalls.

@NelleV:
> I used to reinstall matplotlib every time, but it has become a pain now that the git hash is appended to the version.

I assume you're referring to the thing where pip refuses to install "matplotlib 2.1.dev1+git123" if you already have "matplotlib 2.1.dev1+git456"? Yeah :-(. This is an incredibly frustrating bug that will eventually be fixed :-(. (The pip maintainers at least have finally come around to consensus that it is a bug that needs fixing, but unfortunately the patch got tangled up in the bikeshed around fixing `pip install -U` and is temporarily stalled out.)